_posts/2020-10-08-openfaas-functions-with-github-actions.md

```diff
@@ -0,0 +1,216 @@
+faas-cli is a command line tool used to interact with OpenFaaS, from deploying your functions to viewing logs.
+Follow the installation instructions [here](https://docs.openfaas.com/cli/install/)
+
+> You can follow along with the code used in this tutorial from this [repo](https://github.com/utsavanand2/hello)
+
+
+## Create an OpenFaaS function
+
+Now that we've logged into faasd with faas-cli, we're ready to run our functions.
+I'm going to deploy a golang function for this tutorial.
+
+```sh
+# create a new directory for our project
+mkdir hello
+# pull golang-http templates
+faas-cli template store pull golang-http
+# create a new function with faas-cli
+faas-cli new hello --lang golang-http
+```
+
+If you look into the root directory of the project, faas-cli has created two additional files:
+
+```
+./hello/handler.go
+./hello.yml
+```
+
+The content of our handler looks something like this:
+![./hello/handler.go](/images/2020-10-08-openfaas-functions-with-github-actions/handler-go.jpg)
+
+The YAML file is used to configure the deployment and the build and runtime environment of the function
+![./hello.yml](/images/2020-10-08-openfaas-functions-with-github-actions/hello-yaml.jpg)
+
+## Setup a workflow file for Github Actions
+
+In the root directory of the project run:
+
+```sh
+mkdir -p .github/workflows
+nano ./github/workflows/main.yml
+```
+
+And paste in the following contents into main.yml file
+
+```yaml
+name: CI/CD
+
+on:
+  push:
+    branches: 
+      - master
+jobs:
+  func-build:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Pull template
+        uses: docker://openfaas/faas-cli:latest-root
+        with:
+          args: template store pull golang-http
+      - 
+        name: Run shrinkwrap build
+        uses: docker://openfaas/faas-cli:latest-root
+        with:
+          args: build -f hello.yml --shrinkwrap
+      -
+        name: Login to OpenFaaS Gateway
+        uses: docker://openfaas/faas-cli:latest-root
+        with:
+          args: login -p ${{ secrets.OPENFAAS_GATEWAY_PASSWD }} -g ${{ secrets.OPENFAAS_GATEWAY }}
+      -
+        name: Login to DockerHub
+        if: success()
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - 
+        name: Build and Push the OpenFaaS function
+        uses: docker/build-push-action@v2
+        with:
+          context: ./build/hello/
+          file: ./build/hello/Dockerfile
+          push: true
+          tags: utsavanand2/hello:latest
+      - 
+        name: Deploying the OpenFaaS function
+        uses: docker://openfaas/faas-cli:latest-root
+        with:
+          args: deploy -f hello.yml       
+```
+
+> Note: Replace the Docker UserID from the image tag name with your own.
+
```

Comment:
I think you should add a little more here explaining how/why this works, something like 

> The `faas-cli:latest-root` image has the `faas-cli` installed as the entrypoint, so everything set in `args` is passed to the `faas-cli`.
> This will work with any of the `faas-cli` root tags, you can pin to any specific version of `faas-cli`, for example: `openfaas/faas-cli:0.12.14-root`.

Reply:
Fixed!